duckduckgo/BrowserServicesKit
# Implement simple Subscription Caching mechanism using UserDefaults

**Required**:

Task/Issue URL:  https://app.asana.com/0/1204099484721401/1206574474409573/f
iOS PR:  https://github.com/duckduckgo/iOS/pull/2556
macOS PR: https://github.com/duckduckgo/macos-browser/pull/2359
What kind of version bump will this require?: Minor

**Description**:
- Implements mechanism for caching subscription info
- Caches entitlements
- Cleans cache on signout.
- Returns cached entitlements by default (if present)
- Adds a skipCache parameter to `fetchEntitlements` to force fetching from the service
- Posts a Notification only when entitlements change
- Existing API remains the same.  This does not require changes on local implementation



**Steps to test this PR**:
1.  Purchase a subscription
2. Confirm that entitlements are returned from cache after first use (Breakpoint [here](https://github.com/duckduckgo/BrowserServicesKit/pull/703/files#diff-6024bf9848884f7e6cbb4b48816ebecb139706dd2ee31fb26d08242020aff7f5R268))



---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
